sgx-types/src/ssa.rs

```diff
@@ -270,45 +271,122 @@ pub struct Miscellaneous {
     exinfo: ExceptionInfo,
 }
 
-/// The size of XSAVE region in SSA is derived from the enclave’s support of the collection
-/// of processor extended states that would be managed by XSAVE. The enablement of those
-/// processor extended state components in conjunction with CPUID leaf 0DH information
-/// determines the XSAVE region size in SSA.
-///
-/// Section 38.9, Table 38-7
+/// The legacy region of an XSAVE area comprises the 512 bytes starting at the area’s base address.
+/// See Table 13-1 in
+/// https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-1-manual.pdf
 #[derive(Debug)]
-#[repr(C, align(4096))]
-pub struct XSave([[u64; 32]; 16]);
+#[repr(C)]
+pub struct XSaveLegacy {
+    fcw: u16,
+    fsw: u16,
+    ftw: u8,
+    reserved0: u8,
+    fop: u16,
+    fip31_0: u32,
+    fcs_or_fip47_32: u16,
+    fip63_48_or_reserved: u16,
+    fdp31_0: u32,
+    fds_or_fdp47_32: u16,
+    fdp63_48_or_reserved: u16,
+    mxcsr: u32,
+    mxcsr_mask: u32,
+    mm0: [u8; 10],
+    mm1: [u8; 10],
+    mm2: [u8; 10],
+    mm3: [u8; 10],
+    mm4: [u8; 10],
+    mm5: [u8; 10],
+    mm6: [u8; 10],
+    mm7: [u8; 10],
```

Comment:
I would make a `Mm` type and make this: `mm: [Mm; 8]`. Also, while the individual mm value is 10 bytes, the actual size of this segment is 16 bytes (10 + 6 padding).

You can read about MMX here: https://en.wikipedia.org/wiki/MMX_(instruction_set)

We might want to represent this (internally) as a `u128` with `From` implementations to `u64` that mask off the relevant bits. But we might just want to punt on the `From` implementations altogether.

Reply:
Makes sense; I just saw the padding this morning and added it in, but I'll make a type.

Reply:
@npmccallum I've got `pub struct Mm(u128);` right now to encompass the padding and the values. Were you thinking the `From` would convert between the `u128` and *two* `u64`s to hold the 10 bytes of data?

Reply:
I'd like to avoid the conversions for now. Basically, the size of the register is 80bit x87 floating point. However, I think the top 16 bits are always 1. So it is really a `u64` with some additional reserved space. But I'm not sure of the full implications of that, so I'd like to avoid making a conversion decision until we need to.